src/jemalloc/include/jemalloc/internal/valgrind.h

```diff
@@ -38,9 +38,10 @@
     ptr_maybe_null, old_ptr, old_usize, old_rzsize, old_ptr_maybe_null,	\
     zero) do {								\
 	if (in_valgrind) {						\
+		size_t rzsize = (ptr != NULL) ? p2rz(ptr) : 0;		\
```

Comment:
AFAIR, the problem was not with NULL, but with valgrind instrumentation.
If old_ptr != ptr, but the regions somehow overlap (not possible in original jemalloc), then after valgrind_freelike_block() below, the old region is marked as unacessible and subsequent call to p2rz(ptr) may be interpreted as an access to unallocated memory.
I'm not sure if this is still required in the latest version of this patch - I'll check it out.